source/EvoPlanet/evoplanet.client/src/app/components/simulator/simulator.component.ts

```diff
@@ -1,12 +1,34 @@
-import { Component } from '@angular/core';
+import { HttpClientModule } from '@angular/common/http';
+import { Component, inject, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { SolarSystemService } from '../../services/solar-system/solar-system.service';
+import { IdHandler, SolarSystem } from '../../types';
 
 @Component({
   selector: 'app-simulator',
   standalone: true,
-  imports: [],
+  imports: [HttpClientModule],
   templateUrl: './simulator.component.html',
-  styleUrls: ['./simulator.component.css']
+  styleUrls: ['./simulator.component.css'],
+  providers: [SolarSystemService]
 })
-export class SimulatorComponent {
+export class SimulatorComponent implements OnInit {
+  activatedRoute = inject(ActivatedRoute);
+  handler: IdHandler = { id:'' }
+  router = inject(Router);
+  ngOnInit(): void {
+    const solarId: string = this.activatedRoute.snapshot.params['id'];
+    this.handler.id = solarId
+    this.solarService.getSolarSystemById(this.handler).subscribe(
+      {
+        next: (item) => {
+          this.planetSystem = item;
+        }
+      }
+    );
+  }
+  planetSystem: SolarSystem = { id: ''  };
+  solarService: SolarSystemService = inject(SolarSystemService);
```

Comment:
Putting class members after methods is very confusing here. I suggest moving them up to the other data members, like `router`, etc.

There are cases when data members are pushed to the bottom of a class definition, but it's more prevalent in C# and C++. There are two main approaches to group the content of the class (that I know of):
1. Putting the public members first, privates last (mostly in C#, C++).
2. Putting the data members first, methods last (mostly Java). Public methods still come before private ones.

In this case, separating the members by a method has no justification in any of the above regards.

Reply:
Visual Studio Intellisense put the function up their via my clicks.